crescent/ext/tasks/loop.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+from datetime import timedelta as _timedelta
+from typing import Callable, Sequence, overload
+
+from crescent.ext.tasks.task import Task, TaskCallbackT, link_task
+from crescent.internal import MetaStruct
+
+__all__: Sequence[str] = ("loop", "Loop")
+
+
+class Loop(Task):
+    def __init__(self, callback: TaskCallbackT, timedelta: float) -> None:
+        self.timedelta = timedelta
+        self.first_loop = True
+
+        super().__init__(callback)
+
+    def next_iteration(self) -> float:
+        if self.first_loop:
+            return 0
+        return self.timedelta
+
```

Comment:
Won't this just return the total seconds between loops, no matter how long until the actual next iteration?

Reply:
The docstring for Task.next_iteration says "seconds until next iteration". If I create a loop with a time of 10 seconds, `Loop.next_iteration()` will always return 10s, even if the next iteration is in 1 second.

Reply:
Maybe this function should be private. The functionality is a little weird since the time until next iteration when scheduling doesn't have to be the same as the time until next iteration when the function is called.

Reply:
It's a nice function to have, and it works as expected for the Cron class. Maybe it shouldn't be an abstract method, and only exist on the Cron class? The Loop can use `0 if self.first_loop else self.timedelta`

Reply:
Also that reminds me, `timedelta` is a weird name to use if it's not a timedelta object. Maybe just `delay_seconds`.

Reply:
There 100% has to be an abstract method for this so I don't have to change the implementation of `_call_next`. I can make an extra method on cron fro that though.

Reply:
`_call_next` does the exact same thing. It schedules `_call_async` to be run at some point in the future. That line of code should never be written twice since its fairly complex and the functionality is exactly the same. `Loop` overloads it since thats the best place to say the first iteration of the loop has been scheduled so `delay_seconds` should now be waited for.
`next_iteration` does not say the time until the next function. `next_iteration` returns how long until the process should be scheduled. Both uses of `_next_iteration` are consistent when you use that definition.

Reply:
Hmm. I guess that makes sense. Maybe there should be a docstring on `next_iteration` for `Loop` that clarifies that it will always return 0 or `self.timedelta`? Otherwise it can just be private ig.

Reply:
Ill try to make the docstring better and I think it should be private. This is a hard one to write lol.

Reply:
For the `Loop`, it should just be explicit. "Returns 0 if the first loop has not occurred, otherwise `Loop.timedelta`)

Reply:
good idea